sklearn/impute/tests/test_impute.py

```diff
@@ -351,6 +351,21 @@ def test_imputation_most_frequent_pandas(dtype):
     assert_array_equal(X_trans, X_true)
 
 
+def test_imputation_most_frequent_string_list():
+    X = [['a', 'b'],
+         ['c', np.nan]]
+
+    X_true = np.array([
+        ['a', 'b'],
+        ['c', 'b']
+    ], dtype=np.object)
```

Comment:
numpy has been wanting to [deprecate the use of np.*](https://github.com/numpy/numpy/pull/14882) for while.

```suggestion
    ], dtype=object)
```